File: csharp/flatten-array/FlattenArray.cs
using System.Collections;

public static class FlattenArray
{
    public static IEnumerable Flatten(IEnumerable input)
    {
        foreach (object item in input)
        {
            if (item == null)
                continue;
            IEnumerable? enumerate = (item as IEnumerable);
            if (enumerate != null)
                foreach (object subItem in Flatten(enumerate))
                    yield return subItem;
            else
                yield return item;
        }
    }
}

File: php/flatten-array/FlattenArray.php
<?php
// Flatten array exercise

declare(strict_types=1);

// Flatten an array so any nested arrays have values included inline in the result.
// @param $input: The array to flatten.
// @returns: Array with the flattened value of the array and values from any sub-arrays.
function flatten(array $input): array
{
    $ret = array();

    foreach ($input as $value) {
        if (is_array($value)) {
            $ret = array_merge($ret, flatten($value));
        } else {
            if (!is_null($value)) {
                $ret[] = $value;
            }
        }
    }

    return $ret;
}
